File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    from itertools import combinations
    answer = 0
    
    lst=list(combinations(nums,3))
    
    for n in lst:
        thtn=0
        for i in range(1,(sum(n)//2)+1):
            if sum(n)%i==0:
                thtn+=1
        if thtn==1:
            answer+=1
            
            
    return answer

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;

        int s=0;
        int m=s+1;
        int e=m+1;
        while(s<=nums.length-3){
            int num=nums[s]+nums[m]+nums[e];
         
            if(isPrime(num)) answer++; 
            e++;
            if(e>nums.length-1){
                m++;
                e=m+1;
                if(m>nums.length-2){
                    s++;
                    m=s+1;
                    e=m+1;
                }
            }
            
        }

        return answer;
    }
    
    public static boolean isPrime(int num){
        for(int i=2;i<=Math.sqrt(num);i++){
            if(num%i==0){
                return false;
            }
        }
        return true;
    }
}